### ISSUE TITLE
Typecheck regression in namedarg-hack

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#38392]
[Original report](https://internals.rust-lang.org/t/regression-report-stable-2016-11-08-vs-nightly-2016-12-06/4462/3).

```
   Compiling namedarg_hack v0.1.1474613452 (file:///home/crate)

error[E0308]: mismatched types

  --> src/lib.rs:22:44

   |

22 |             if item.ident.name.as_str() == "hack_plugin_registrar" {

   |                                            ^^^^^^^^^^^^^^^^^^^^^^^ expected struct `syntax::symbol::InternedString`, found reference

   |

   = note: expected type `syntax::symbol::InternedString`

   = note:    found type `&'static str`


error: aborting due to previous error


error: Could not compile `namedarg_hack`.
```

cc @alexcrichton 

This is a _plugin_ that seems to compile successfully on 1.14 beta but not stable!

**Comment by alexcrichton** [comment]
Oh looks like this crate detects whether it's compiling against nightly. If not, it compiles an empty crate, and if so, it compiles the crate. This means that crater detected a success on beta (an empty compile) and a compile failure on nightly because we changed the compiler at some point.

tl;dr; this is just a "regression" in the sense that we changed the compiler, so it's an expected regression, and not one we'll want to track. Closing.
